src/components/Carousel/styles.ts

```diff
@@ -0,0 +1,42 @@
+import { StyleSheet } from "react-native";
+import GlobalStyles from "../../utils/styleGlobal";
+
```

Comment:
no meu ver nao é necessario nenhum flex e achei bem loco essas subtrações negativas hahaha o problema que vc estava tendo é o que comentei da flatlist o tamanho da imagem estava vindo preenchendo e depois era aplicado o witdth o mesmo na resolução da imagem era como se a imagem divesse um background gigante, solução foi força um maxHeight na FlatList vou deixar aqui em baixo como deixei o index e o style

###### index.tsx 
```
import React, { useState } from 'react'

import {
  FlatList,
  Text,
  View,
  Image,
  ScrollView,
  useWindowDimensions,
} from 'react-native'

import { styles } from './styles'

type CarouselProps = {
  data?: Array<{
    id: string
    img: string
  }>
  name: string
  description: string
}

export const Carousel = ({ data, name, description }: CarouselProps) => {
  const { width } = useWindowDimensions()
  const [selectedIndex, setSelectedIndex] = useState(0)

  const Circle = ({ selected }: { selected: boolean }) => {
    const style = {
      backgroundColor: selected ? '#DD736C' : '#fff',
    }

    return <View style={[styles.circle, style]} />
  }

  return (
    <View style={styles.container}>
      <FlatList
        style={styles.image}
        data={data}
        horizontal
        showsHorizontalScrollIndicator={false}
        pagingEnabled
        keyExtractor={(item) => item.id}
        onMomentumScrollEnd={(event) => {
          const index =
            event.nativeEvent.contentOffset.x /
            event.nativeEvent.layoutMeasurement.width
          setSelectedIndex(Math.round(index))
        }}
        renderItem={(item: any) => {
          return (
            <Image
              source={{
                uri: item.item.img,
              }}
              style={{ width: width }}
            />
          )
        }}
      />
      <View style={styles.circlesContainer}>
        {data?.map((img, index) => (
          <Circle key={index} selected={index === selectedIndex} />
        ))}
      </View>

      <Text style={styles.namePart}>{name}</Text>
      <ScrollView style={styles.scroll}>
        <Text style={styles.descriptionPart}>{description}</Text>
      </ScrollView>
    </View>
  )
}

```
###### styles.ts

```
import { StyleSheet } from 'react-native'
import { GlobalStyles } from '../../utils/styleGlobal'

export const styles = StyleSheet.create({
  container: {
    backgroundColor: GlobalStyles.$primary_black,
    marginBottom: 15,
  },
  image: {
    maxHeight: 320,
  },
  namePart: {
    fontSize: GlobalStyles.$typography23,
    color: GlobalStyles.$primary_red,
    margin: 15,
  },
  scroll: {
    height: 200,
  },
  descriptionPart: {
    fontSize: GlobalStyles.$typography16,
    color: GlobalStyles.$primary_white,
    marginLeft: 15,
    marginRight: 15,
    marginBottom: 15,
  },
  circle: {
    width: 6,
    height: 6,
    borderRadius: 3,
    marginHorizontal: 3,
  },
  circlesContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
})

```